wgpu/src/util/mod.rs

```diff
@@ -76,17 +87,46 @@ pub fn make_spirv_raw(data: &[u8]) -> Cow<'_, [u32]> {
 
     // Before checking if the data starts with the magic, check if it starts
     // with the magic in non-native endianness, own & swap the data if so.
-    if words[0] == MAGIC_NUMBER.swap_bytes() {
+    if words[0] == SPIRV_MAGIC_NUMBER.swap_bytes() {
         for word in Cow::to_mut(&mut words) {
             *word = word.swap_bytes();
         }
     }
 
-    assert_eq!(
-        words[0], MAGIC_NUMBER,
-        "wrong magic word {:x}. Make sure you are using a binary SPIRV file.",
```

Comment:
Unfortunately, we lose the ability to output the magic number here, since formatting doesn't work in constants. While there are workarounds that work with the current MSRV like the `const_panic` crate, I decided that this isn't a massive loss and it was better to not add more dependencies.